python/tank/platform/validation.py

```diff
@@ -285,9 +285,15 @@ def __validate_schema_value(self, settings_key, schema):
         data_type = schema.get("type")
         self.__validate_schema_type(settings_key, data_type)
 
-        if "default_value" in schema:
+        # Get a list of all keys that start with the default value key string.
+        # Some types, like hooks, allow for engine specific default values such
+        # as "default_value_tk-maya". Validate each of these key's values
+        # against the setting's data type.
+        default_value_keys = [k for k in schema if k.startswith("default_value")]
```

Comment:
From what I could tell in the code, these special `default_value_{engine_name}` keys aren't specific to hooks types. Please correct me if I'm wrong. 

FYI, This particular section didn't technically need to be modified for sparse configs, I just noticed that it wasn't validating the engine-specific default values. So I guess this could cause breakage for folks with invalid, but never used, engine-specific defaults in their manifests. Let me know if you want to not do this.